### PR TITLE
store GroupResult as_tuple()

### PR DESCRIPTION
**Pull request by AllexVeldman**
Store the entire group as serialised by `as_tuple()` instead of the list of Task id's.

This will also change the return value for `_restore_group()["result"]` from `List[AsyncResult]` to `GroupResult` ensuring 
```python
group = GroupResult(group_id)
group.save()
group == <backend>.restore_group(group_id=group_id)
```

### CODE REVIEW

**Comment by auvipy** [comment]
should we wait for this or cut a new release from master?

**Comment by AllexVeldman** [comment]
@auvipy I don't have the time this week to properly test this and compare it with the `GroupResult.restore()` from the redis backend. As the restore functionality is new, this feature might just be broken if you release from master now. it will unblock upgrading to celery 5.1.0 so I think you should go for it.

**Comment by auvipy** [comment]
yes possible, i will check

**Comment by AllexVeldman** [comment]
@auvipy I think this should be good to go, I reverted the change in the ChordCounter for now to minimize the scope to just the bugfix

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
I will try to release a new version today

**Pull request merged**
